src/core/include/openvino/core/parallel.hpp

```diff
@@ -441,25 +457,30 @@ template <typename T0, typename F>
 void parallel_for(const T0& D0, const F& func) {
 #if OV_THREAD == OV_THREAD_TBB
     auto work_amount = static_cast<size_t>(D0);
-    int nthr = parallel_get_max_threads();
-    if (static_cast<size_t>(nthr) > work_amount)
-        nthr = static_cast<int>(work_amount);
-    if (nthr == 1) {
+    const int nthr = parallel_get_max_threads();
+    int virtual_threads = nthr;
+    if (tbb_partitioner == TBB_AUTO) {
+        virtual_threads = 1 == nthr ? 1 : nthr * MULTIPLIER;
+    }
```

Comment:
```suggestion
    int nthr = parallel_get_max_threads();    
    if (tbb_partitioner == TBB_AUTO && 1 != nthr) {
        nthr *= MULTIPLIER;
    }
```
It looks like `virtual_threads` is redundant, consider rename `nthr` if name is not good enough.